### PR TITLE
[Feat] Add tpu-testing in circleci

### PR DESCRIPTION
**Pull request by edgarriba**
### Description

This PR enables CircleCI to work with GCP-TPU and torch-xla.

We only implement kornia color for now. A set of incremental PRs will be sent to expand the functionality.

### Status
**Ready**

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or new feature that would cause existing functionality to change)
- [x] New tests added to cover the changes
- [ ] Docstrings/Documentation updated

### PR Implementer
This is a small checklist for the implementation details of this PR.

If there are any questions regarding code style or other conventions check out our 
[summary](https://github.com/kornia/kornia/blob/master/CONTRIBUTING.rst).

- [ ] Did you discuss the functionality or any breaking changes before ?
- [ ] **Pass all tests**: did you test in local ? `make test`
- [ ] Unittests: did you add tests for your new functionality ?
- [ ] Documentations: did you build documentation ? `make build-docs`
- [ ] Implementation: is your code well commented and follow conventions ? `make lint`
- [ ] Docstrings & Typing: has your code documentation and typing ? `make mypy`
- [ ] Update notebooks & documentation if necessary

### KorniaTeam
<details>
  <summary>KorniaTeam workflow</summary>
  
  - [ ] Assign correct label
  - [ ] Assign PR to a reviewer
  - [ ] Does this PR close an Issue? (add `closes #IssueNumber` at the bottom if 
        not already in description)

</details>

### Reviewer
<details>
  <summary>Reviewer workflow</summary>

  - [ ] Do all tests pass? (Unittests, Typing, Linting, Documentation, Environment)
  - [ ] Does the implementation follow `kornia` design conventions?
  - [ ] Is the documentation complete enough ?
  - [ ] Are the tests covering simple and corner cases ?
 
</details>

### CODE REVIEW

**Comment by zcain117** [comment]
> @zcain117 I'm getting the following error which I guess is an error of the pytorch-xla version:
> 
> ```
> ImportError while loading conftest '/kornia/conftest.py'.
> conftest.py:5: in <module>
>     import torch
> /root/miniconda3/envs/container/lib/python3.6/site-packages/torch/__init__.py:189: in <module>
>     _load_global_deps()
> /root/miniconda3/envs/container/lib/python3.6/site-packages/torch/__init__.py:142: in _load_global_deps
>     ctypes.CDLL(lib_path, mode=ctypes.RTLD_GLOBAL)
> /root/miniconda3/envs/container/lib/python3.6/ctypes/__init__.py:348: in __init__
>     self._handle = _dlopen(self._name, mode)
> E   OSError: libmkl_intel_lp64.so: cannot open shared object file: No such file or directory
> ```

Hm very strange, your Dockerfile looks very similar to ours, which was working today: https://github.com/GoogleCloudPlatform/ml-testing-accelerators/blob/master/ci_pytorch/Dockerfile

There is a chance that the Kornia dependencies are causing some mismatch of torch + torch_xla versions, which is why we see this error sometimes. I read through the [logs](https://app.circleci.com/pipelines/github/kornia/kornia/1951/workflows/19138b5c-f21f-4f8d-aee2-aab73f9f63ed/jobs/4777/parallel-runs/0/steps/0-118) of your build but didn't see anything obvious

Just as an experiment, could you try skipping the Kornia install and running one of our repo's tests like this example: https://github.com/GoogleCloudPlatform/ml-testing-accelerators/blob/master/ci_pytorch/tpu_test.jsonnet

File: conftest.py [hunk 1]
```diff
@@ -43,7 +43,14 @@ def get_test_dtypes() -> Dict[str, torch.dtype]:
...
-    if device_name not in TEST_DEVICES:
+    # To work with TPUs on Colab:
+    # pip install cloud-tpu-client==0.10
+    # pip install https://storage.googleapis.com/tpu-pytorch/wheels/torch_xla-1.6-cp36-cp36m-linux_x86_64.whl
+    # Issue: gradcheck on TPUs takes ages...
+    if device_name == 'tpu':
+        import torch_xla.core.xla_model as xm
+        return xm.xla_device()
+    elif device_name not in TEST_DEVICES:
         pytest.skip(f"Unsupported device type: {device_name}")
```
Comment (shijianjian) [inline conftest.py:54]: Should this return None?

Comment (shijianjian) [reply conftest.py:54]: Try something like:
```bash
$ pytest -vv -r sxf test/ --device tpu --dtype float64
```
to show the skipped reason. I would suspect it falls into ```pytest.skip(f"Unsupported device type: {device_name}")```

**Comment by edgarriba** [comment]
@zcain117 @shijianjian I think I know the issue. The jsonnet file takes somehow the kornia git master which contains the conftest.py for pytest.

**Comment by edgarriba** [comment]
@zcain117 I think it's somehow working :) 
A couple of questions: what's the preferred dtype for xla devices ? and I see few issues around torchscript , it's supposed to be supported ?

**Comment by edgarriba** [comment]
@shijianjian this is ready for color module. We can incrementally start adding the other modules. I'm working now with enhance.

@zcain117 thanks for everything:)
BTW,  I tried wheels for pytorch-xla 1.6/1.7 and it's full of errors. I only managed to make it work for nightlies.

**Pull request merged**